### PR TITLE
Cloudwatch: Add linting to restrict imports from core

### LINKED ISSUE #78845: CloudWatch: Enable lint rules to prevent importing from core grafana
There are linting rules that will ensure that the plugin does not import Grafana core code (and that will not add new restricted import in the future). To enable them:

- [ ] Frontend:
  - Add the plugin directory to [.eslintrc](https://github.com/grafana/grafana/blob/main/.eslintrc#L99-L100)
  - Run `yarn lint`
    - Note that the new linter is a bit more strict, it may return errors related to the import order but that can be easily fixed with the `--fix` option.
    - If you missed some core imports, the linter will detect them and raise errors.
- [ ] Backend:
  - Add the plugin directory to [.golangci.toml](https://github.com/grafana/grafana/blob/main/.golangci.toml#L57-L58)
  - Run `make lint-go`

### PR DESCRIPTION
**Pull request by idastambuk**
**What is this feature?**

Final step of removing FE + BE dependencies on core grafana. Enables linting for CW to prevent core grafana imports

**Which issue(s) does this PR fix?**:

Fixes https://github.com/grafana/grafana/issues/78845

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: pkg/tsdb/cloudwatch/cloudwatch.go [hunk 1]
```diff
@@ -122,7 +122,7 @@ func NewInstanceSettings(httpClientProvider *httpclient.Provider) datasource.Ins
 	}
 }
 
-// cloudWatchExecutor executes CloudWatch requests.
+// cloudWatchExecutor executes CloudWatch requests
```
Comment (idastambuk) [inline pkg/tsdb/cloudwatch/cloudwatch.go:125]: to trigger BE checks

**Review by gabor — APPROVED** [review]
eslint-part is fine 👍

File: .golangci.toml [hunk 2]
```diff
@@ -67,6 +67,7 @@ files = [
   "**/pkg/tsdb/parca/**/*",
   "**/pkg/tsdb/tempo/*",
   "**/pkg/tsdb/tempo/**/*",
+   "**/pkg/tsdb/cloudwatch/**/*",
```
Comment (gabor) [inline .golangci.toml:70]: i don't know how exactly this thing works, but other datasources have 2 lines each... cloudwatch only 1. is this intentional?

File: .eslintrc [hunk 3]
```diff
@@ -115,7 +115,8 @@
         "public/app/plugins/datasource/loki/*.{ts,tsx}",
         "public/app/plugins/datasource/loki/**/*.{ts,tsx}",
         "public/app/plugins/datasource/elasticsearch/*.{ts,tsx}",
-        "public/app/plugins/datasource/elasticsearch/**/*.{ts,tsx}"
+        "public/app/plugins/datasource/elasticsearch/**/*.{ts,tsx}",
+        "public/app/plugins/datasource/cloudwatch/**/*.{ts,tsx}"
```
Comment (kevinwcyu) [inline .eslintrc:119]:
```suggestion
        "public/app/plugins/datasource/cloudwatch/*.{ts,tsx}",
        "public/app/plugins/datasource/cloudwatch/**/*.{ts,tsx}"
```

File: .golangci.toml (hunk 2 above)
Comment (kevinwcyu) [inline .golangci.toml:70]:
```suggestion
  "**/pkg/tsdb/cloudwatch/*",
  "**/pkg/tsdb/cloudwatch/**/*",
```

**Review by kevinwcyu — APPROVED** [review]

File: .golangci.toml (hunk 2 above)
Comment (idastambuk) [reply .golangci.toml]: Fixed!

**Pull request merged**